cephbackupstorage/cephbackupstorage/cephagent.py

```
@@ -1072,6 +1127,175 @@ def cancel(self, req):
+        shell.call('tar -zxf %s -C %s' % (install_path, unzipInstallPath))
+        unzipFiles = {}
+        for f in os.listdir(unzipInstallPath):
+            unzipFiles[os.path.join(unzipInstallPath, f)] = os.path.getsize(os.path.join(unzipInstallPath, f))
+        return unzipInstallPath, unzipFiles
+
+    @replyerror
+    def get_upload_file_progress(self, req):
+        cmd = jsonobject.loads(req[http.REQUEST_BODY])
+        task = self.upload_tasks.get_task(cmd.taskUuid)
+        if task is None:
+            raise Exception('file not found')
```

Ruff (0.14.14)

[warning] Create your own exception

(TRY002)